Code/Framework/AzToolsFramework/AzToolsFramework/ViewportUi/ViewportUiDisplay.cpp

```diff
@@ -361,11 +373,34 @@ namespace AzToolsFramework::ViewportUi::Internal
         m_fullScreenLayout.addLayout(&m_uiOverlayLayout, 0, 0, 1, 1);
 
         // format the label which will appear on top of the highlight border
-        AZStd::string styleSheet = AZStd::string::format("background-color: %s; border: none;", HighlightBorderColor);
+        styleSheet = AZStd::string::format("background-color: %s; border: none;", HighlightBorderColor);
         m_viewportBorderText.setStyleSheet(styleSheet.c_str());
         m_viewportBorderText.setFixedHeight(ViewportUiTopBorderSize);
         m_viewportBorderText.setVisible(false);
         m_fullScreenLayout.addWidget(&m_viewportBorderText, 0, 0, Qt::AlignTop | Qt::AlignHCenter);
+
+        // format the back button which will appear in the top right of the highlight border
+        styleSheet = AZStd::string::format(
+            "border: 0px; padding-left: %dpx; padding-right: %dpx", HighlightBorderBackButtonMargin, HighlightBorderBackButtonMargin);
+        m_viewportBorderBackButton.setStyleSheet(styleSheet.c_str());
+        m_viewportBorderBackButton.setVisible(false);
+        m_viewportBorderBackButton.setText(HighlightBorderBackButtonText);
+
+        // setup the handler for the back button to call the user provided callback (if any)
+        QObject::connect(
+            &m_viewportBorderBackButton, &QPushButton::clicked,
+            [this]()
+            {
+                if (m_viewportBorderBackButtonCallback.has_value())
+                {
+                    // we need to swap out the existing back button callback because it will be reset in RemoveViewportBorder()
+                    AZStd::optional<ViewportUIBackButtonCallback> backButtonCallback;
+                    m_viewportBorderBackButtonCallback.swap(backButtonCallback);
+                    RemoveViewportBorder();
+                    (*backButtonCallback)();
```

Comment:
question: Aaah is this to handle lifetime issues where the thing destroys itself, we had this issue with Manipulators at one point too, might be worth explicitly mentioning 'lifetime' in this comment

Reply:
fixed 👍